kivymd/uix/button/button.py

```diff
@@ -755,92 +795,128 @@ def on_touch_down(self, touch):
+    and defaults to `' '`.
+    """
 
-class BaseFlatButton(BaseRectangularButton):
-    def update_md_bg_color(
-        self, instance_theme_manager, name_palette: str
-    ) -> NoReturn:
-        """Called when the application color palette changes."""
+    _min_width = NumericProperty(88)
 
-        self.text_color = self.theme_cls.primary_color
 
-    def set_text_color(self, interval: Union[int, float]) -> NoReturn:
-        """Sets the text color if no custom value is specified."""
+class ButtonContentsIcon():
+    """Contents for a round BaseButton consisting of an MDIcon."""
 
-        if self.text_color in ([0.0, 0.0, 0.0, 0.87], [1, 1, 1, 1]):
-            self.theme_text_color = "Custom"
-            self.text_color = self.theme_cls.primary_color
+    icon_size = NumericProperty()
+    """
+    Icon font size.
+    Use this parameter as the font size, that is, in sp units.
 
-    def on_md_bg_color(self, instance_button, color: list) -> NoReturn:
-        """
-        We override this method, thus prohibiting setting the background color
-        for the button.
+    .. versionadded:: 1.0.0
 
-        Allows to set the background color only in the rangefrom
-        [0.0, 0.0, 0.0, 0.0] to [0.0, 0.0, 0.0, 0.1]. This color is set in
-        the :class:`BasePressedButton` class when the button is pressed and
-        Ignore other custom colors.
-        """
+    :attr:`icon_size` is an :class:`~kivy.properties.NumericProperty`
+    and defaults to `None`.
+    """
 
-        if color[:-1] != [0.0, 0.0, 0.0]:
-            self.md_bg_color = [0.0, 0.0, 0.0, 0.0]
+    icon = StringProperty("checkbox-blank-circle")
+    """
+    Button icon.
 
-    def on_disabled(self, instance_button, value_disabled: bool) -> NoReturn:
-        if value_disabled and not self.disabled:
-            self.md_bg_color = (
-                self.md_bg_color_disabled
-                if self.md_bg_color_disabled
-                else self.theme_cls.disabled_hint_text_color
-            )
-        else:
-            if self._md_bg_color:
-                self.md_bg_color = self._md_bg_color
+    :attr:`icon` is an :class:`~kivy.properties.StringProperty`
+    and defaults to `'checkbox-blank-circle'`.
+    """
 
-    def on_elevation(self, instance_button, elevation_value: int) -> NoReturn:
-        """
-        We are overriding this method to not allow set the `elevation` value
-        for this type of button.
-        """
+    icon_color = ColorProperty(None)
+    """
+    Button icon color.
 
-        self.elevation = 0
-        self._elevation = 0
-        super().on_elevation(instance_button, elevation_value)
+    :attr:`icon_color` is an :class:`~kivy.properties.ColorProperty`
+    and defaults to `None`.
+    """
+    _icon_color = ColorProperty(None)
+
+    def set_text_color(self, *args):
+        super().set_text_color()
+        self._icon_color = self.icon_color or self._text_color
+
+
+class ButtonContentsIconText():
```

Comment:
Why is this class not inherited from any widget?

Reply:
It's just a mix-in; it shouldn't be instantiated on it's own but just adds contents to a BaseButton.

I would like to move to a better and more flexible system anyway for adding content to buttons - or possibly move to a simpler system that only supports one or zero icons and one or zero text labels for all buttons.